sparkler-app/src/main/scala/edu/usc/irds/sparkler/pipeline/FetchFunction.scala

```diff
@@ -17,55 +17,51 @@
 
 package edu.usc.irds.sparkler.pipeline
 
-import java.io.ByteArrayOutputStream
-import java.net.{HttpURLConnection, URL}
 import java.util.Date
 
+import edu.usc.irds.sparkler.{Fetcher}
 import edu.usc.irds.sparkler.base.Loggable
 import edu.usc.irds.sparkler.model.ResourceStatus._
-import edu.usc.irds.sparkler.model.{Content, Resource}
+import edu.usc.irds.sparkler.model.{FetchedData, SparklerJob, Content, Resource}
+import edu.usc.irds.sparkler.service.PluginService
 import org.apache.nutch.metadata.Metadata
 
 import scala.language.postfixOps
 
 /**
   * Created by thammegr on 6/7/16.
   */
-object FetchFunction extends ((Resource) => Content) with Serializable with Loggable {
+object FetchFunction extends ((SparklerJob, Resource) => Content) with Serializable with Loggable {
 
   val FETCH_TIMEOUT = 1000
 
-  override def apply(resource: Resource): Content = {
+  override def apply(job: SparklerJob, resource: Resource): Content = {
     LOG.info("FETCHING {}", resource.url)
-    //FIXME: this is a prototype, make it real
-    //TODO: handle errors
+    //TODO: Fetcher Plugin Integrated. Improve on this. Handle Errors
+
     val fetchedAt = new Date()
     val metadata = new Metadata()
     try {
-      val urlConn = new URL(resource.url).openConnection()
-      urlConn.setConnectTimeout(FETCH_TIMEOUT)
-
-      val responseCode = urlConn.asInstanceOf[HttpURLConnection].getResponseCode
-      LOG.debug("STATUS CODE : " + responseCode + " " + resource.url)
+      val fetcher:scala.Option[Fetcher] = PluginService.getExtension(classOf[Fetcher], job)
+      var fetchedData: FetchedData = fetcher.get.fetch(resource.url)
```

Comment:
ideally, we shouldn't call `Option.get`. Instead we should use `match` pattern and tell how to deal with `null/None`

See how Rahul has correctly handled this [here]( https://github.com/USCDataScience/sparkler/blob/8ccf8088d5fcccc7a8dc5a466553da8753cfbad5/sparkler-app/src/main/scala/edu/usc/irds/sparkler/pipeline/Crawler.scala#L168-L170)

In our case when fetcher is None, we shall fall back to DefaultFetcher